src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -0,0 +1,86 @@
+#include "pch.h"
+
+#include "CTerminalHandoff.h"
+
+// The callback function when a connection is received
+static NewHandoff _pfnHandoff = nullptr;
+// The registration ID of the class object for clean up later
+static DWORD g_cTerminalHandoffRegistration = 0;
+
+// Routine Description:
+// - Starts listening for TerminalHandoff requests by registering
+//   our class and interface with COM.
+// Arguments:
+// - pfnHandoff - Function to callback when a handoff is received
+// Return Value:
+// - S_OK, E_NOT_VALID_STATE (start called when already started) or relevant COM registration error.
+HRESULT CTerminalHandoff::s_StartListening(NewHandoff pfnHandoff) noexcept
+try
+{
+    RETURN_HR_IF(E_NOT_VALID_STATE, _pfnHandoff != nullptr);

```

Comment:
Huh, there's no `RETURN_HR_IF_NOT_NULL`?

Reply:
No, there isn't. I just checked. There is a `RETURN_HR_IF_NULL_EXPECTED` but that does the same as `RETURN_HR_IF_NULL` just without having a fit on the debug print channel.